sklearn/neighbors/_classification.py

```diff
@@ -67,8 +67,11 @@ class KNeighborsClassifier(KNeighborsMixin, ClassifierMixin, NeighborsBase):
     metric : str or callable, default='minkowski'
         The distance metric to use for the tree.  The default metric is
         minkowski, and with p=2 is equivalent to the standard Euclidean
-        metric. For a list of available metrics, see the documentation of
-        :class:`~sklearn.metrics.DistanceMetric`.
+        metric. 
+        For a list of available metrics, see the documentation of
+        :class:`~sklearn.metrics.DistanceMetric` and the metrics listed in
+        `~sklearn.metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS`. Note that the
+        "cosine" metric uses :func:`~sklearn.metrics.pairwise.cosine_distances`.
```

Comment:
There is a [linting error](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=36472&view=logs&j=32e2e1bb-a28f-5b18-6cfc-3f01273f5609&t=fc67071d-c3d4-58b8-d38e-cafc0d3c731a). We can fix it as so:

```suggestion
        metric. For a list of available metrics, see the documentation of
        :class:`~sklearn.metrics.DistanceMetric` and the metrics listed in
        `sklearn.metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS`. Note that the
        "cosine" metric uses :func:`~sklearn.metrics.pairwise.cosine_distances`.
```

Note there is no `~` in front of `PAIRWISE_DISTANCE_FUNCTIONS` because there is no actual link to `PAIRWISE_DISTANCE_FUNCTIONS`.

This comment applies to the other diffs in this PR as well.